pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -406,6 +408,20 @@ class AnalyticsImpl implements Analytics {
 
     _logHandler.save(data: body);
 
+    // Conditional logic for clearing contents of persisted
+    // files (except for config file) on opt out
+    if (!reportingBool) {
+      _sessionHandler.sessionFile.writeAsStringSync('');
+      _logHandler.logFile.writeAsStringSync('');
+      _clientIdFile.writeAsStringSync('');
```

Comment:
I would have assumed that the files would be deleted from disk, not just emptied. Is there a reason for leaving an empty file on disk?

Reply:
We asked to have it be cleared because I didn't want to change the initializer logic, on startup, the `Initializer` class checks to make sure each of these files exist, if not, it will create them. And this happens before we determine if the user has opted out so it would be a lot more refactoring